kedro/framework/cli/starters.py

```diff
@@ -353,8 +351,8 @@ def new(  # noqa: PLR0913
         # If interactive flow used, print hint
         interactive_flow = prompts_required and not config_path
         _print_selection_and_prompt_info(
-            extra_context.get("tools"),
-            extra_context.get("example_pipeline"),
+            extra_context["tools"],
+            extra_context["example_pipeline"],
```

Comment:
Why did you change it from `.get()` to this direct access? I always feel like `.get()` is the safer way in case the key doesn't exist in the dictionary.

Reply:
Yes, using .get() is indeed a safer method for retrieving values from a dictionary. However, I noticed we were overusing it in every step of our processing. This made the code less precise and harder to comprehend, particularly in discerning whether and where default values were being provided.
Upon analyzing the code, I realized the default values were actually set in the _get_extra_context() function. During the refactoring of this function, I made the inclusion of default values more explicit by introducing a 'default' section and provided comments about it in DocString. Subsequently, I replaced all instances of variable access using get() with direct access. This change aims to enhance the clarity and strictness of our code.
Now, I hope it's clear that values must be specified, and to understand which values are used, one should review the previous processing steps instead of worrying about default values at each stage.